client/components/app.jsx

```diff
@@ -11,10 +12,13 @@ export default class App extends React.Component {
       message: null,
       isLoading: true,
       view: { name: 'catalog', params: {} },
-      cart: []
+      cart: [],
+      total: 0
```

Comment:
Since the total can be calculated from the cart items, you don't need to store it in state.